rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/telemetry/TbMsgTimeseriesNode.java

```diff
@@ -111,7 +141,9 @@ public void onMsg(TbContext ctx, TbMsg msg) {
                 .entityId(msg.getOriginator())
                 .entries(tsKvEntryList)
                 .ttl(ttl)
-                .saveLatest(!config.isSkipLatestPersistence())
+                .saveTimeseries(saveTimeseries)
```

Comment:
replace three boolean variables with a single java object (rename PersistenceDecision -> TimeseriesSaveRequestSettings or similar and use it).